bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/handler/BlueZBridgeHandler.java

```diff
@@ -167,70 +112,36 @@ private void startDiscovery() {
     }
 
     private void refreshDevices() {
-        try {
+        refreshTry: try {
             logger.debug("Refreshing Bluetooth device list...");
             List<tinyb.BluetoothDevice> tinybDevices = adapter.getDevices();
             logger.debug("Found {} Bluetooth devices.", tinybDevices.size());
             for (tinyb.BluetoothDevice tinybDevice : tinybDevices) {
                 BlueZBluetoothDevice device = getDevice(new BluetoothAddress(tinybDevice.getAddress()));
                 device.updateTinybDevice(tinybDevice);
-                notifyDiscoveryListeners(device);
-            }
-            // clean up orphaned entries
-            synchronized (devices) {
-                for (BlueZBluetoothDevice device : devices.values()) {
-                    if (shouldRemove(device)) {
-                        logger.debug("Removing device '{}' due to inactivity", device.getAddress());
-                        device.dispose();
-                        devices.remove(device.getAddress());
-                        discoveryListeners.forEach(listener -> listener.deviceRemoved(device));
-                    }
-                }
+                deviceDiscovered(device);
             }
             // For whatever reason, bluez will sometimes turn off scanning. So we just make sure it keeps running.
             startDiscovery();
-            // everything went fine, so lets switch to online
-            updateStatus(ThingStatus.ONLINE);
         } catch (BluetoothException ex) {
             String message = ex.getMessage();
             if (message != null) {
                 if (message.contains("Operation already in progress")) {
                     // we shouldn't go offline in this case
-                    return;
+                    break refreshTry;
```

Comment:
Nice. Didn't know about that.

Reply:
I was surprised when I first found out about that too.
Using labels allows you to break out of any code block.
I rarely find the need to use it though but it does come in handy on occasion.